Core/include/Acts/Visualization/Interpolation3D.hpp

```diff
@@ -0,0 +1,57 @@
+// This file is part of the ACTS project.
+//
+// Copyright (C) 2016 CERN for the benefit of the ACTS project
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+#pragma once
+
+#include "Acts/Definitions/Algebra.hpp"
+
+#include <unsupported/Eigen/Splines>
+
+namespace Acts::Interpolation3D {
+
+/// @brief Helper function to interpolate points using a spline
+/// from Eigen
+///
+/// @tparam input_vector_type
+/// @param inputs input vector points
+/// @param nPoints number of interpolation points
+///
+/// @return std::vector<Acts::Vector3> interpolated points
+template <typename input_vector_type>
+std::vector<Acts::Vector3> spline(const std::vector<input_vector_type>& inputs,
```

Comment:
I wonder if we should fully template this on a range to decouple from `std::vector`? Also do we plan to use this with something else than `Vector3` as input?

Reply:
Absolutely, if I want to user it with the actsvg plugin, I will need another input/output type.

I made it agnostic to it.